src/java/main/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -390,6 +391,7 @@ else if (qp.getType() == Leader.SNAP) {
                             + Long.toHexString(qp.getZxid()));
                     System.exit(13);
                 }
+                zk.getZKDatabase().setlastProcessedZxid(qp.getZxid());
```

Comment:
@rakeshadr Created https://issues.apache.org/jira/browse/ZOOKEEPER-2833 to avoid this gets lost.